### ISSUE TITLE
Configure  `active_record` as part of `rails`

### ISSUE THREAD

**Issue by yijia-zhan** [DataDog/dd-trace-rb#764]
With 

```ruby
Datadog.configure do |c|
  c.tracer env: Rails.env
  c.use :rails, service_name: "myapp-rails"
end
```
I see the following services instrumented in Datadog:
```
- active_record
- myapp-rails
- myapp-rails-cache
- myapp-rails-postgres
```

I understand there's `c.use :active_record` and `c.use :rails`, but I'm not sure why `active_record` is configured separately from `rails`. What I want is to have:
```
- myapp-active_record
- myapp-rails
- myapp-rails-cache
- myapp-rails-postgres
```

Is there a way to achieve this? I read this issue https://github.com/DataDog/dd-trace-rb/issues/574 and this PR https://github.com/DataDog/dd-trace-rb/pull/450, and it seems that the following is exactly what I need:

```ruby
Datadog.configure do |c|
  c.use :rails do |rails|
    rails.service_name = 'myapp'
    
    rails.active_record do |active_record|
      active_record.orm_service_name = 'myapp-active_record'
    end
  end
end
```
But with this I get a `NoMethodError`:
```
undefined method `active_record' for #<Datadog::Contrib::Rails::Configuration::Settings:0x00007ff12073d9d0> (NoMethodError)
```

**Comment by yijia-zhan** [comment]
If I do the following:

```ruby
Datadog.configure do |c|
  c.tracer env: Rails.env
  c.use :active_record, orm_service_name: "myapp-active_record"
  c.use :rails, service_name: "myapp-rails"
end
```

I seem to be able to get what I want. Just want to clarify, is this going to "double register" `active_record`? Does the order of `c.use :active_record` and `c.use :rails` matter?

**Comment by delner** [comment]
@yijia-zhan don't worry, it won't double the instrumentation; if you ever call `use` twice, it will only apply the new settings the second time.

Sorry about the confusion of looking at that other PR. We intend to implement that `rails.active_record` block configuration style, however, we aren't quite ready for it just yet. Hopefully in the near future that syntax will work just fine.

**Comment by yijia-zhan** [comment]
Thanks @delner! Just want to double confirm on whether the order of `c.use :active_record` and `c.use :rails` matters. 

If I do:
```ruby
Datadog.configure do |c|
  c.use :active_record, service_name: "test", orm_service_name: "myapp-active_record"
  c.use :rails, service_name: "myapp-rails"
end
```
Would the `test` service name for `active_record` be overridden by the the one specified as part of rails? (https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/rails/framework.rb#L57-L65)

On the other hand, if I do not specify a `service_name` for `c.use :active_record`, would it default to the one specified as part of rails?

**Comment by delner** [comment]
It's a great question, and largely depends on the load order of your application. I presume this configuration is in an initializer. Rails configuration is setup to apply after the application is fully initialized, so there's a decent chance that `c.use :rails` will override ActiveRecord with its own service name, regardless of order. I would have to test this to verify.

Obviously that's not ideal behavior (part of the reason we wanted to introduce that block syntax), but I think the following should work as a work around:

```ruby
Datadog.configure do |c|
  c.use :active_record, orm_service_name: "myapp-active_record"
  c.use :rails, service_name: "myapp-rails", database_service: "test"
end
```

The Rails integration, when it reconfigures ActiveRecord will pass `database_name` as `service_name` to ActiveRecord. Give that a shot, see if that works for you.

**Comment by yijia-zhan** [comment]
Thanks @delner. Will test out and report back.

**Comment by delner** [comment]
Hey @yijia-zhan, just wanted to let you know we're in the process of implementing the block syntax for Rails, as you shared in your original post. I'll keep you posted on any developments!

**Comment by yijia-zhan** [comment]
This is very good to know! Thank you for the update @delner !

**Comment by delner** [comment]
Still a WIP but the PR for this is here: #810

**Comment by caius** [comment]
We ran into this today, I found setting another after_initialize block and setting the `orm_service_name` directly on the `active_record` integration configuration appears to look identical to defining the two integrations and having one override the other.

```ruby
# frozen_string_literal: true

require "ddtrace"

Datadog.configure do |c|
  c.version = App::GIT_REVISION

  c.use :rails, service_name: "app-web", cache_service: "app-cache", database_service: "app-postgres"
end

Rails.application.config.after_initialize do
  # Workaround for :rails not setting :orm_service_name on :active_record https://github.com/DataDog/dd-trace-rb/issues/764
  if (active_record_integration = Datadog.configuration.instrumented_integrations[:active_record])
    active_record_integration.configuration[:orm_service_name] = "app-active_record"
  end
end
```

**Comment by agirlnamedsophia** [comment]
How does all this work with v1 whereby `orm_service_name` has been removed? What should folks upgrading do in this case? Thank you!

**Comment by marcotc** [comment]
👋 @agirlnamedsophia, `orm_service_name` only affected `active_record.instantiation` spans.
We removed this configuration because ActiveRecord object instantiation is part of the host application workload, thus in v1.0.0 the service for `active_record.instantiation` is the same as the global application `DD_SERVICE` (or `c.service = ...`).

Let us know if you have a specific use case that needed `orm_service_name` before that doesn't work anymore in v1.0.0.
I looked it up and the `active_record.instantiation` spans will generate standalone metrics you can monitor and graph on, as [they are considered "measured", as well as having the option to configure `analytics_enabled: true`](https://github.com/DataDog/dd-trace-rb/blob/b8801d15bc25ab64abee941ef61fc2457104f1f0/lib/datadog/tracing/contrib/active_record/events/instantiation.rb#L40-L46) if you need them analyzed.

**Comment by delner** [comment]
Yes, as @marcotc said, in theory you simply remove `orm_service_name` and that's it.

You shouldn't really expect any major changes to occur beyond that, especially as it concerns the database spans themselves.

**Comment by agirlnamedsophia** [comment]
Thanks @marcotc and @delner for responding! We removed `orm_service_name` and I helped the engineers see how to find those spans with the `operation` dropdown. We are good to go!

**Comment by frsantos** [comment]
Hi @delner. Is the `active_record` supposed to be used on all the integrations? I see that Rails is using it, but Shoryuken is not reporting any active record instantiations, only database queries.

I have something like the following

```ruby
Datadog.configure do |c|
  c.tracing.instrument :rails,         service_name: "#{prefix}Rails", middleware_names: true
  c.tracing.instrument :rack,          service_name: "#{prefix}Rails", request_queuing: true, web_service_name: "#{prefix}nginx"
  c.tracing.instrument :active_record, service_name: "#{prefix}ActiveRecord"
  c.tracing.instrument :shoryuken,     service_name: "#{prefix}Shoryuken"
  ...
end
```
